### PR TITLE
mitosheet: add routes for in-app upgrade

### PR DESCRIPTION
**Pull request by naterush**
# Description

Follows this spec: https://www.notion.so/trymito/In-App-Upgrading-to-Pro-cebc0f863a1e4594b8c5982aa88667c8

# Testing

Test the in-app changes. Do so by 1) searching for what you would expect, and 2) adding those search terms if they don't appear. Then, just test the taskpane. 

Also, test the new command line interface in `mitosheet` for going pro - you just need to do `python -m mitosheet turnonpro <ACCESS CODE>`.

# Documentation

N/A.

### CODE REVIEW

File: mitosheet/mitosheet/user/utils.py [hunk 1]
```diff
@@ -91,3 +93,7 @@ def is_excel_import_enabled() -> bool:
     pandas_version_valid = not is_prev_version(pd.__version__, '0.25.0')
 
     return python_version_valid and pandas_version_valid
+
+
+def check_pro_acccess_code(access_code: Optional[str]) -> bool:
```
Comment (naterush) [inline mitosheet/mitosheet/user/utils.py:98]: Doc string!

File: mitosheet/src/components/icons/PurpleCheckMark.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        <svg width="18" height="15" viewBox="0 0 18 15" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path d="M2 8.88525L6.15622 12.4914C6.57338 12.8534 7.20496 12.8086 7.5669 12.3914L16.3562 2.26144" stroke="#9D6CFF" strokeWidth="4"/>
+        </svg>
+    )
+}
+
+export default PurpleCheckMark;
+
+
+
```
Comment (naterush) [inline mitosheet/src/components/icons/PurpleCheckMark.tsx:16]: Remove

File: mitosheet/src/components/taskpanes/UpgradeToPro/UpgradeToProTaskpane.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                    </p>
+                    <ProListElement text="Turn off all telemetry"/>
+                    <ProListElement text="Generated code optimization"/>
+                    <ProListElement text="Style graphs"/>
+                    <ProListElement text="Export formatting"/>
+                    <ProListElement text="Priority support"/>
+                    <ProListElement text="All future pro functionality!"/>
+                    <div className="mt-25px">
+                        <Row justify="space-around">
+                            <form className='element-width-block' action="https://jl76z192i0.execute-api.us-east-1.amazonaws.com/Prod/create_checkout_session/" method="POST" target="_blank">
```
Comment (naterush) [inline mitosheet/src/components/taskpanes/UpgradeToPro/UpgradeToProTaskpane.tsx:71]: Comment this

File: mitosheet/src/hooks/useInputValue.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { useState } from "react";
+
+/* 
+    This effect supplies an input with a value and an
```
Comment (naterush) [inline mitosheet/src/hooks/useInputValue.tsx:4]: Comment

File: mitosheet/src/utils/pro.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+// Utilities for the Open Source version of Mito that deal with Mito Pro
+
+// If the user signs up for pro, this is the access code they must put in
```
Comment (naterush) [inline mitosheet/src/utils/pro.tsx:3]: Remove

**Review by naterush — COMMENTED** [review]
Cleanup

File: mitosheet/src/components/taskpanes/UpgradeToPro/UpgradeToProTaskpane.tsx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    if (!isPro && !isEnteringAccessCode) {
+        return (
+            <DefaultTaskpane>
+                <DefaultTaskpaneHeader 
+                    header="Upgrade to Mito Pro"
+                    setUIState={props.setUIState}            
+                />
+                <DefaultTaskpaneBody>
+                    <p className="text-body-1 mb-10px">
+                        <a href='https://trymito.io/plans' target='_blank' rel="noreferrer"><span className='text-body-1-link'>Mito Pro</span></a> gives you extra features to super charge you analysis:
```
Comment (aarondr77) [inline mitosheet/src/components/taskpanes/UpgradeToPro/UpgradeToProTaskpane.tsx:61]: super charge you -> super charge your

**Comment by aarondr77** [comment]
Can we move this ? to the left. 

Screen Shot 2022-04-20 at 3 02 39 PM

File: mitosheet/src/components/taskpanes/UpgradeToPro/UpgradeToProTaskpane.tsx [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+                    <ProListElement text="Priority support"/>
+                    <ProListElement text="All future pro functionality!"/>
+                    <div className="mt-25px">
+                        <Row justify="space-around">
+                            {/* The Stripe link is no normal link, and requires a POST, so we're required to use a form to allow users to click on it */}
+                            <form className='element-width-block' action="https://jl76z192i0.execute-api.us-east-1.amazonaws.com/Prod/create_checkout_session/" method="POST" target="_blank">
+                                <button 
+                                    className={classNames('text-button', 'text-header-3', 'text-overflow-wrap', 'element-width-block', 'text-button-variant-dark')} 
+                                    type="submit"
+                                    onClick={() => {
```
Comment (aarondr77) [inline mitosheet/src/components/taskpanes/UpgradeToPro/UpgradeToProTaskpane.tsx:76]: Can we add logging to this so we can tell that users are upgrading via this?

**Review by aarondr77 — COMMENTED** [review]
Nice. It's all working well. 

The only other thing that we might want to do here is update the docs a bit. Do we want to get rid of the Installing Mito Pro section now and have everyone go through the access code?  I'm good with whatever you decide (or if we decide to do it later)

**Comment by naterush** [comment]
Let's update the docs later, when we move to actual CTAs in the app to upgrade. I think it'll make the most sense then - but good call. 

Thanks for the review! Merging when tests pass :-)

**Pull request merged**